openpype/hosts/houdini/api/pipeline.py

```diff
@@ -35,71 +40,139 @@
 
 
-self = sys.modules[__name__]
-self._has_been_setup = False
-self._parent = None
-self._events = dict()
-
-
-def install():
-    _register_callbacks()
-
-    pyblish.api.register_host("houdini")
-    pyblish.api.register_host("hython")
-    pyblish.api.register_host("hpython")
-
-    pyblish.api.register_plugin_path(PUBLISH_PATH)
-    register_loader_plugin_path(LOAD_PATH)
-    register_creator_plugin_path(CREATE_PATH)
-
-    log.info("Installing callbacks ... ")
-    # register_event_callback("init", on_init)
-    register_event_callback("before.save", before_save)
-    register_event_callback("save", on_save)
-    register_event_callback("open", on_open)
-    register_event_callback("new", on_new)
-
-    pyblish.api.register_callback(
-        "instanceToggled", on_pyblish_instance_toggled
-    )
-
-    self._has_been_setup = True
-    # add houdini vendor packages
-    hou_pythonpath = os.path.join(HOUDINI_HOST_DIR, "vendor")
-
-    sys.path.append(hou_pythonpath)
-
-    # Set asset settings for the empty scene directly after launch of Houdini
-    # so it initializes into the correct scene FPS, Frame Range, etc.
-    # todo: make sure this doesn't trigger when opening with last workfile
-    _set_context_settings()
-    shelves.generate_shelves()
-
-
-def uninstall():
-    """Uninstall Houdini-specific functionality of avalon-core.
-
-    This function is called automatically on calling `api.uninstall()`.
-    """
+class HoudiniHost(HostBase, IWorkfileHost, ILoadHost, INewPublisher):
+    name = "houdini"
+
+    def __init__(self):
+        super(HoudiniHost, self).__init__()
+        self._op_events = {}
+        self._has_been_setup = False
+
+    def install(self):
+        pyblish.api.register_host("houdini")
+        pyblish.api.register_host("hython")
+        pyblish.api.register_host("hpython")
+
+        pyblish.api.register_plugin_path(PUBLISH_PATH)
+        register_loader_plugin_path(LOAD_PATH)
+        register_creator_plugin_path(CREATE_PATH)
+
+        log.info("Installing callbacks ... ")
+        # register_event_callback("init", on_init)
+        self._register_callbacks()
+        register_event_callback("before.save", before_save)
+        register_event_callback("save", on_save)
+        register_event_callback("open", on_open)
+        register_event_callback("new", on_new)
+
+        pyblish.api.register_callback(
+            "instanceToggled", on_pyblish_instance_toggled
+        )
+
+        self._has_been_setup = True
+        # add houdini vendor packages
+        hou_pythonpath = os.path.join(HOUDINI_HOST_DIR, "vendor")
+
+        sys.path.append(hou_pythonpath)
+
+        # Set asset settings for the empty scene directly after launch of
+        # Houdini so it initializes into the correct scene FPS,
+        # Frame Range, etc.
+        # TODO: make sure this doesn't trigger when
+        #       opening with last workfile.
+        _set_context_settings()
+        shelves.generate_shelves()
```

Comment:
Somehow I can't start Houdini with this line present. As soon as I remove this `shelves.generate_shelves` then it launches fine. With this line in the codebase Houdini just appears to hang during launch on the splash screen with the console only saying:
```
Installing OpenPype ...
Setting scene FPS to 25
```
And I have no idea why this happening.
Tested with Houdini 19.5.303 Py 3.7 on Windows.

Seems odd no?

Reply:
There is strange logic, from my point of view. It crashes with `FileNotFoundError` instead of warning log - I don't think shelves are so crutial to crash. Maybe that's the issue.

There is in default settings empty shelf item which could cause triggering the error
![image](https://user-images.githubusercontent.com/43494761/192274178-63a08b9f-3649-4ce2-be17-25afcb3124de.png)

Reply:
It didn't crash. It just didn't launch. Note that I was using Houdini core!